frontend/src/features/env/SwitchEnvFeedbackModal.tsx

```diff
@@ -0,0 +1,185 @@
+    },
+    [submitSwitchEnvFormFeedbackMutation],
+  )
+
+  const handleFormSubmit = handleSubmit((inputs) => {
+    submitFeedback(inputs)
+    setShowThanksPage(true)
+  })
+
+  const handleClose = () => {
+    onClose()
+    setShowThanksPage(false)
+  }
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={handleClose}
+      size={modalSize}
+      initialFocusRef={initialRef}
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalCloseButton />
+        {showThanksPage ? (
+          <>
+            <ModalHeader>Thank you for your feedback!</ModalHeader>
+            <ModalBody>
+              Would you like to switch back to the original FormSG?
+            </ModalBody>
+            <ModalFooter mt={{ base: '2.5rem', md: '0' }}>
+              <Stack
+                w="100%"
+                spacing="1rem"
+                justify="right"
+                direction={{ base: 'column-reverse', md: 'row' }}
+              >
+                <Button
+                  isFullWidth={isMobile}
+                  variant="clear"
+                  onClick={() => {
+                    onClose()
+                    setShowThanksPage(false)
+                  }}
+                >
+                  No, don’t switch
+                </Button>
+                <Button
+                  isFullWidth={isMobile}
+                  onClick={() => {
+                    user
+                      ? adminSwitchEnvMutation.mutate()
+                      : publicSwitchEnvMutation.mutate()
+                    onClose()
+                    setShowThanksPage(false)
+                  }}
+                >
+                  Yes, please
+                </Button>
+              </Stack>
+            </ModalFooter>
+          </>
+        ) : (
+          <chakra.form noValidate onSubmit={handleFormSubmit}>
+            <ModalHeader>Something not right on the new FormSG?</ModalHeader>
+            <ModalBody mt="0" pt="0">
+              <Stack spacing="1rem">
+                <FormControl>
+                  <Input type="hidden" {...register('url')} value={url} />
+                </FormControl>
+                <FormControl>
+                  <FormLabel
+                    description={
+                      user
+                        ? ''
+                        : 'Any fields you’ve filled in your form so far will be cleared'
+                    }
+                  >
+                    Please tell us what we can improve on
+                  </FormLabel>
+                  <Textarea {...register('feedback')} tabIndex={1} />
+                </FormControl>
+                <FormControl>
+                  {user ? (
+                    <Input
+                      type="hidden"
+                      {...register('email')}
+                      value={user.email}
+                    />
+                  ) : (
+                    <></>
+                  )}
+                </FormControl>
```

Comment:
/nit: is the `FormControl` wrapper necessary when `user` is falsy? Also, shouldn't we use `null` rather than the empty markup? (am not sure whats the recommended approach is)

```suggestion
                { user ? (
                  <FormControl>
                    <Input
                      type="hidden"
                      {...register('email')}
                      value={user.email}
                    />
                  </FormControl>
                  ) : null }
                
```

Reply:
you're right `FormControl` wrapper isn't needed when `user` is falsy, amended